### PR TITLE
Rustdoc: don't hide anonymous reexport

### LINKED ISSUE #108931: rustdoc doesn't show traits re-exported as `_`
I tried this code:

```rust
mod my_mod {
    pub trait Foo {}
}

pub mod prelude {
    pub use super::my_mod::Foo as _;
}
```

I expected to see this happen: rustdoc shows that the prelude re-exports

Instead, this happened: rustdoc shows the module as empty

This is a regression from https://github.com/rust-lang/rust/pull/97617, which explicitly implements the above behaviour (CC @GuillaumeGomez). This makes it hard to figure out from the documentation which kinds of traits importing a prelude-style module would bring into scope unless the prelude re-exports them with an actual name. Re-exporting with an actual name is not necessary for extension traits though.

### Meta

`rustc --version --verbose`:
```
rustc 1.69.0-nightly (001a77fac 2023-01-30)
binary: rustc
commit-hash: 001a77fac33f6560ff361ff38f661ff5f1c6bf85
commit-date: 2023-01-30
host: x86_64-unknown-linux-gnu
release: 1.69.0-nightly
LLVM version: 15.0.7
```

@rustbot modify labels: regression-untriaged

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rust-lang/rust/issues/108931.

From https://github.com/rust-lang/rust/issues/108931, it appears that having anonymous re-exports for traits is actually used in some places, so instead of hiding them automatically, we should prevent them to be ever inlined.

r? @notriddle

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
I approve both suggestions (only for traits and the compiler warning).

**Comment by sdroege** [comment]
The reference says https://doc.rust-lang.org/reference/items/use-declarations.html#underscore-imports

> Items can be imported without binding to a name by using an underscore with the form use path as _. This is particularly useful to import a trait so that its methods may be used without importing the trait's symbol, for example if the trait's symbol may conflict with another symbol. Another example is to link an external crate without importing its name.

So crates might also have an effect, but which? Together with `#[macro_use]` maybe?

This should be in a separate PR in any case, and in either case should probably also get some documentation updates so the behaviour of it for non-trait items is actually clear.

**Comment by GuillaumeGomez** [comment]
I'm fine with this approach too. If so, the current PR is ready then.

File: src/librustdoc/visit_ast.rs [hunk 1]
```diff
@@ -330,7 +335,7 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
                 }
             }
             // If we're inlining, skip private items or item reexported as "_".
```
Comment (aDotInTheVoid) [inline src/librustdoc/visit_ast.rs:337]: This comment probably needs to be updated

Comment (GuillaumeGomez) [reply src/librustdoc/visit_ast.rs]: Good catch!

**Pull request merged**